src/credentials/DeviceAttestationVerifier.h

```diff
@@ -198,23 +198,35 @@ class DeviceAttestationVerifier
     DeviceAttestationVerifier(const DeviceAttestationVerifier &) = delete;
     DeviceAttestationVerifier & operator=(const DeviceAttestationVerifier &) = delete;
 
+    struct AttestationInfo
+    {
+        AttestationInfo(const ByteSpan & attestationElementsBuffer, const ByteSpan & attestationChallengeBuffer,
+                        const ByteSpan & attestationSignatureBuffer, const ByteSpan & paiDerBuffer, const ByteSpan & dacDerBuffer,
+                        const ByteSpan & attestationNonce, VendorId vendorId, uint16_t productId) :
+            mAttestationElementsBuffer(attestationElementsBuffer),
+            mAttestationChallengeBuffer(attestationChallengeBuffer), mAttestationSignatureBuffer(attestationSignatureBuffer),
+            mPaiDerBuffer(paiDerBuffer), mDacDerBuffer(dacDerBuffer), mAttestationNonce(attestationNonce), mVendorId(vendorId),
+            mProductId(productId)
+        {}
+        const ByteSpan &
+            mAttestationElementsBuffer; // Buffer containing attestation elements portion of Attestation Response (raw TLV)
+        const ByteSpan & mAttestationChallengeBuffer; // Buffer containing the attestation challenge from the secure session
+        const ByteSpan & mAttestationSignatureBuffer; // Buffer the signature portion of Attestation Response
+        const ByteSpan & mPaiDerBuffer;               // Buffer containing the PAI certificate from device in DER format.
+        const ByteSpan & mDacDerBuffer;               // Buffer containing the DAC certificate from device in DER format.
+        const ByteSpan & mAttestationNonce;           // Buffer containing attestation nonce.
```

Comment:
Suggest using values of `ByteSpan` rather than const refs, so that the spans have the same backing even if the spans are stored in a context where the *original span with same data* is out of scope. Subtle, but important for the async API. Otherwise we have 2 scopes to care about: the scope where AttestationInfo object exists (which usually points to the stack), and the scope of the backing of the above data (which usually points to storage).

ByteSpans are small objects, so having them by value is fine.